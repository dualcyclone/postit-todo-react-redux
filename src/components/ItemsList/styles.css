.itemsList-ul {
  list-style: none;
}

.itemsList-ul {
  overflow: hidden;
  padding: 3em;
}
