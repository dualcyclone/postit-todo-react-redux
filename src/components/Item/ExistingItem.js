import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { completeItem, removeItem } from '../../main/store/actions'
import Item from './Item'
import './styles.css'

export const ExistingItem = ({ item, onDoneClick, onRemoveClick }) => (
  <Item
    className={ item.completed ? 'complete' : 'todo' }
    color={ item.color }>
      <p>{ item.content }</p>
      <a onClick={onDoneClick} className="icon icono-cross">&nbsp</a>
      <a onClick={onDoneClick} className="icon icono-check">&nbsp</a>
      <a onClick={onRemoveClick} className="icon icono-trash">&nbsp</a>
  </Item>
)

ExistingItem.propTypes = {
  item: PropTypes.object.isRequired,
  onDoneClick: PropTypes.func.isRequired,
  onRemoveClick: PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch, ownProps) => ({
  onDoneClick: () => dispatch(completeItem(ownProps.item)),
  onRemoveClick: () => dispatch(removeItem(ownProps.item))
})

export default connect(
  null,
  mapDispatchToProps
)(ExistingItem)
