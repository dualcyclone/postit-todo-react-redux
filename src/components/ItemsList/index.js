import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Item from '../Item';
import './styles.css';

const ItemsList = ({ items }) => {
  return (
    <div>
      { items.length === 0 &&
        <p id={'items-missing'}>
          Add some tasks above.
        </p>
      }
      { items.length > 0 &&
        <ul className={'itemsList-ul'}>
          { items.map(item => (
            <Item key={item.id} item={item}>{ item.content }</Item>
          )) }
        </ul>
      }
    </div>
  );
}

ItemsList.propTypes = {
  items: PropTypes.array.isRequired,
};

const mapStateToProps = (state) => {
  return {
    items: state.items
  };
};

export default connect(
  mapStateToProps
)(ItemsList);
