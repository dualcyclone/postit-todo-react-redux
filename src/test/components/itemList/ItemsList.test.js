import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import { ItemsList } from '../../../../src/components/ItemsList/index';
import ExistingItem from '../../../../src/components/Item/ExistingItem';

const defaultProps = {
  items: [],
};

describe('ItemsList', () => {
  it('Should render without failing', () => {
    shallow(<ItemsList {...defaultProps} />);
  });

  it('Should not display a ul if there arent any items to render', () => {
    const renderedItem = shallow(
      <ItemsList {...defaultProps} items={[]} />
    );

    expect(renderedItem.find('ul')).to.have.length(0);
  });

  it('Should render ExistingItem components as child components when there are valid items to render', () => {
    const items = [
      { id: 1, content: 'Test 1' },
      { id: 2, content: 'Test 2' }
    ];

    const renderedItem = shallow(
      <ItemsList {...defaultProps} items={items} />
    );
    
    expect(renderedItem.find('ul').children(ExistingItem)).to.have.length(2);
  });
});
