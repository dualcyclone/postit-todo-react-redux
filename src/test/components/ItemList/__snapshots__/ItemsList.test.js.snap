exports[`ItemsList Should render without failing 1`] = `<div />`;
